pkg/image/association.go

```diff
@@ -116,36 +210,43 @@ func (as *Associations) Decode(r io.Reader) error {
 		return fmt.Errorf("error decoding image associations: %v", err)
 	}
 	// Update paths for local usage.
-	for k, assoc := range *as {
-		assoc.Path = filepath.FromSlash(assoc.Path)
-		(*as)[k] = assoc
+	for _, imageName := range as.Keys() {
+		assocs, _ := as.Search(imageName)
+		for _, assoc := range assocs {
+			assoc.Path = filepath.FromSlash(assoc.Path)
+			as.UpdateValue(imageName, assoc)
+		}
 	}
 	return nil
 }
 
 func (as Associations) validate() error {
 	var errs []error
-	for _, a := range as {
-		if s, ok := imageTypeStrings[a.Type]; ok && s != "" {
-			continue
-		}
-		switch a.Type {
-		case TypeInvalid:
-			// TypeInvalid is the default value for the concrete type, which means the field was not set.
-			errs = append(errs, fmt.Errorf("image %q: must set image type", a.Name))
-		default:
-			errs = append(errs, fmt.Errorf("image %q: unknown image type %v", a.Name, a.Type))
-		}
+	for _, imageName := range as.Keys() {
+		assocs, _ := as.Search(imageName)
```

Comment:
This error should be checked right?

Reply:
That is actually a boolean, but I could just do the check and generate a new error if the key isn't found in the map.